### PR TITLE
updated error message for failing to find pods before port forward

### PR DESCRIPTION
**Pull request by TheRealSibasishBehera**
Signed-off-by: sibashi <fangedhamster3114@gmail.com>
 
  * while doing port-forward , if pod is not found , the error message is always 
           `Error: kubearmor pod not found`
           
           
Now the new error message also specifies the pod that was supposed to be found
For example in case of `karmor summary` if the client fails to find to find the pod
  
`Error: kubearmor discovery-engine pod not found`

### CODE REVIEW

File: discover/discover.go [hunk 1]
```diff
@@ -42,14 +42,15 @@ var port int64 = 9089
 // ConvertPolicy converts the knoxautopolicies to KubeArmor and Cilium policies
 func ConvertPolicy(c *k8s.Client, o Options) error {
 	gRPC := ""
+	targetPodName := "discovery-engine"
```
Comment (nyrahul) [inline discover/discover.go:45]: lets call this targetSvc
```suggestion
	targetSvc := "discovery-engine"
```

File: utils/portforward.go [hunk 2]
```diff
@@ -22,20 +22,22 @@ import (
...
+	RemotePort    int64
+	MatchLabels   map[string]string
+	Namespace     string
+	PodName       string
+	TargetPodName string
 }
 
 // InitiatePortForward : Initiate port forwarding
-func InitiatePortForward(c *k8s.Client, localPort int64, remotePort int64, matchLabels map[string]string) (PortForwardOpt, error) {
+func InitiatePortForward(c *k8s.Client, localPort int64, remotePort int64, matchLabels map[string]string, targetPodName string) (PortForwardOpt, error) {
```
Comment (nyrahul) [inline utils/portforward.go:34]:
```suggestion
func InitiatePortForward(c *k8s.Client, localPort int64, remotePort int64, matchLabels map[string]string, targetSvc string) (PortForwardOpt, error) {
```

File: utils/portforward.go [hunk 3]
```diff
@@ -22,20 +22,22 @@ import (
...
 	pf := PortForwardOpt{
-		LocalPort:   localPort,
-		RemotePort:  remotePort,
-		Namespace:   "",
-		MatchLabels: matchLabels,
+		LocalPort:     localPort,
+		RemotePort:    remotePort,
+		Namespace:     "",
+		MatchLabels:   matchLabels,
+		TargetPodName: targetPodName,
```
Comment (nyrahul) [inline utils/portforward.go:40]:
```suggestion
		TargetSvc: targetSvc,
```

File: utils/portforward.go [hunk 4]
```diff
@@ -118,7 +120,7 @@ func (pf *PortForwardOpt) getPodName(c *k8s.Client) error {
 		return err
 	}
 	if len(podList.Items) == 0 {
-		return errors.New("kubearmor pod not found")
+		return fmt.Errorf("kubearmor %s pod not found", pf.TargetPodName)
```
Comment (nyrahul) [inline utils/portforward.go:123]:
```suggestion
		return fmt.Errorf("%s svc not found", pf.TargetSvc)
```

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: discover/discover.go (hunk 1 above)
Comment (TheRealSibasishBehera) [reply discover/discover.go:45]: Yeah , will change `PodName` to `Svc`

**Review by nyrahul — APPROVED** [review]

**Pull request merged**